### ISSUE TITLE
Relaxing product name validation rules

### ISSUE THREAD

**Issue by tswfi** [PrestaShop/PrestaShop#25791]
#### Is your feature request related to a problem?

The current validation for product name (and category name and some other places too) contain check for some special characters like `<>;=#{}` (https://github.com/PrestaShop/PrestaShop/blob/c30432393755fd0e55cf1b43ef72546e29fc9a64/classes/Validate.php#L365)

This causes problems if product name contains `=` character for example.

After a brief discussion on slack with @matks it looks like these are really **old** restrictions that have probably been made to avoid some xss issues. 

The checks are not even that complete, they don't check the length of the string so the validation might give ok result but the data cannot be saved into db as it is too long.

#### Describe the solution you'd like

Remove the character checks and do a length check instead.

#### Alternatives you've considered

Move all the checks from separate "validation" and "form" checks into symfony database object layer. Each object knows the fields and the restrictions and could generate the required validation checks for forms to use. This would remove validation duplication (for example here the validation is "hardcoded" into the form https://github.com/PrestaShop/PrestaShop/blob/c30432393755fd0e55cf1b43ef72546e29fc9a64/src/PrestaShopBundle/Form/Admin/Product/ProductInformation.php#L218 
and if the rules change on db level they need to be changed here also.

Also for some reason there is min 3 char check here: https://github.com/PrestaShop/PrestaShop/blob/c30432393755fd0e55cf1b43ef72546e29fc9a64/src/PrestaShopBundle/Form/Admin/Product/ProductInformation.php#L222 
Even if two character product names are not really that common I don't think there should be a check that prevents merchant from creating a product named "OK" for example.

#### Additional context

There are some other checks that could be cleaned also. 

For example customer first and last name and dots:
![image](https://user-images.githubusercontent.com/1352132/132177728-cef42000-3095-4794-ac18-4f0b283d9a65.png)
That should probably be removed also see the old classic: https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

**Comment by matks** [comment]
As mentioned, my opinion is that indeed these checks are outdated and need to be improved 😄 

> Remove the character checks and do a length check instead.

However you suggest to remove it all ? 😄 

There is one thing we discussed: maybe we should at least verify that input can be considered a valid slug?

**Comment by tswfi** [comment]
> However you suggest to remove it all ? 

Nah, I suggest that the length should be checked not the characters :)

> There is one thing we discussed: maybe we should at least verify that input can be considered a valid slug?

The input string does not need to be a valid slug. `slugify` method takes care of converting the input string to a slug.
 https://github.com/PrestaShop/PrestaShop/blob/9723602bfddb7240c52c1720e6ecac837d07be6e/classes/lang/DataLang.php#L117

But yes, the slug generation from string need to be tested that it still produces correct slugs with new characters (I really hope it already checks stuff like this `      test <> string   ###  ` which probably should produce `test-string`

**Comment by matks** [comment]
> For example customer first and last name and dots:

For this one there is a historical reason: https://build.prestashop.com/news/prestashop-1-7-5-2-1-6-1-24-maintenance-release/ (PR https://github.com/PrestaShop/PrestaShop/pull/13604)

**Comment by hibatallahAouadni** [comment]
Hello @tswfi 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by saulaski** [comment]
> The input string does not need to be a valid slug. slugify method takes care of converting the input string to a slug.

Agreed, SEO constraints should be handled where they belong and not by the catalog of products.
But such a major change could be impacting, and the final decision belongs to product managers.

**Comment by tswfi** [comment]
Just an update to this ancient thing. At least product name validation rules have been relaxed at some point. in 9.0.1 I happened to test this right now allows lots more chars. From the list below only these were blocked: `<>{}` (not sure why but they are :)

<img width="890" height="171" alt="Image" src="https://github.com/user-attachments/assets/9d82a762-182b-4a11-bf4e-66fb8970816d" />

and the length check is also gone from product name.

Didn't check other things but might be that this could even be closed as "not applicable anymore" (or something like that).

**Comment by boo-code** [comment]
Confirming @tswfi's 2026 update, and the second half of the request has landed too, so both parts of this are addressed on `9.2.x`.

The character list is gone. `Validate::isCatalogName()` is now the whole check:

```php
public static function isCatalogName($name)
{
    return preg_match('/^[^<>{}]*$/u', $name);
}
```

So `<>{}` are the only four left, which matches what you measured on 9.0.1.

The length check asked for in the description also exists now, and as one value rather than duplicated: `ProductSettings::MAX_NAME_LENGTH = 128` matches the `ps_product_lang.name` column, `varchar(128)`, and it is applied on both paths - `'size' => ProductSettings::MAX_NAME_LENGTH` in `Product::$definition`, and `new Length(['max' => ProductSettings::MAX_NAME_LENGTH])` in `HeaderType.php:89`. That removes the "validation passes but the row will not fit" case from the original report.

On why those four remain, and this is a reading rather than a citation: `<` and `>` are the usual defence in depth against markup, and `{` and `}` matter because a product name reaches mail templates, where `Mail::Send()` substitutes `{token}` placeholders with `strtr()` on the finished HTML. A name carrying braces is the one input that can collide with that mechanism, so relaxing those two is not the same size of change as dropping `;=#` was.
